modules/nlu/src/backend/nlu-service.ts

```diff
@@ -0,0 +1,101 @@
+import axios, { AxiosInstance } from 'axios'
+import * as sdk from 'botpress/sdk'
+import { NLU } from 'botpress/sdk'
+
+import { LegacyIntentService } from './intents/legacy-intent-service'
+import { LegacyIntentDefinition } from './typings'
+
+export class NLUService {
+  private _isLegacy: boolean
+  private _botId: string
+  private _client: AxiosInstance
+
+  constructor(
+    private bp: typeof sdk,
+    botConfig: sdk.BotConfig,
+    private legacyIntentService: LegacyIntentService,
+    private logger: NLU.Logger
+  ) {
+    this._isLegacy = !botConfig.oneflow
+    this._botId = botConfig.id
+  }
+
+  public async getIntentsAndEntities(): Promise<{
+    intentDefs: NLU.IntentDefinition[]
+    entityDefs: NLU.EntityDefinition[]
+  }> {
+    this._client = axios.create(await this.bp.http.getAxiosConfigForBot(this._botId, { localUrl: true }))
```

Comment:
Just so you know that is quite common when dealing with authenticated clients, what we usually see is :
create an instance with a token (axiosConfig in your case) then optimistically try to fetch data. if you have a 401 then refresh token and try to fetch again.

Reply:
Ok, should I implement this?